app/src/main/java/com/example/bookwormadventuresdeluxe2/BookListAdapter.java

```diff
@@ -54,31 +59,52 @@ public BookListAdapter(Context context, FirestoreRecyclerOptions options)
         return bookListViewHolder;
     }
 
-    @Override
-    protected void onBindViewHolder(@NonNull BookListViewHolder holder, int position, @NonNull Book book)
+    private View.OnClickListener launchDetailView(DetailView bookDetailFragment, Book book, String documentId)
     {
-        /* Set the text on the item view for each book */
-        String documentId = getSnapshots().getSnapshot(position).getId();
-        holder.title.setText(book.getTitle());
-        holder.author.setText(book.getAuthor());
-        holder.isbn.setText(book.getIsbn());
-
-        book.setStatusCircleColor(book.getStatus(), holder.statusCircle);
-
-        /* We need to continually update the onClick listener in the case of filtered queries */
-        holder.itemView.setOnClickListener(new View.OnClickListener()
-        {
+        View.OnClickListener listener = new View.OnClickListener() {
             // Handles a click on an item in the recycler view
             @Override
             public void onClick(View v)
             {
                 // Opens the book in detail view
-                MyBooksDetailViewFragment bookDetailFragment = new MyBooksDetailViewFragment();
                 bookDetailFragment.onFragmentInteraction(book, documentId);
 
                 ((MyBooksActivity) context).getSupportFragmentManager().beginTransaction()
                         .replace(R.id.frame_container, bookDetailFragment).commit();
             }
-        });
+        };
+        return listener;
+    }
+
+    @Override
+    protected void onBindViewHolder(@NonNull BookListViewHolder holder, int position, @NonNull Book book)
+    {
+        // Set the text on the item view for each book
+        String documentId = getSnapshots().getSnapshot(position).getId();
+        holder.title.setText(book.getTitle());
+        holder.author.setText(book.getAuthor());
+        holder.isbn.setText(book.getIsbn());
+        DetailView detailView;
+        String user = UserCredentialAPI.getInstance().getUserId();
+        book.setStatusCircleColor(holder.statusCircle, user);
+
+        switch (this.caller)
+        {
+            case R.id.my_books:
+                detailView = new MyBooksDetailViewFragment();
+                break;
+            case R.id.requests:
+                detailView = new RequestDetailViewFragment();
+                break;
+            case R.id.borrow:
+                detailView = new BorrowDetailViewFragment();
+                break;
+            default:
+                Log.d("Error", "Error in BookListAdapter: caller not found");

```

Comment:
Is this left behind from debugging? Does it need to be throw?

Reply:
Could change it to throw an error. It trips when an unknown Fragment id is calling DetailView. I set it to act the same as MyBooksFragment because I don't think it needs to error.

Reply:
What are the consequences of it falling down to default? If it is going to be a major concern (i.e. some fragment changed its name and now one of our buttons is unresponsibe) we should change it to be an exception

Reply:
default will mean that when clicking on a booklist item whose adapter wasnt properly implemented it will act the same as the clicking on a booklist item on MyBooksFragment (will open the book in MyBooksDetailView)